app/src/main/java/com/github/h3lp3rs/h3lp/AwaitHelpActivity.kt

```diff
@@ -1,41 +1,184 @@
+        val latLng = LatLng(latitude, longitude)
+        val name = "Helper coming to help you" //TODO RESOURCES
+
+        val options = MarkerOptions()
+        options.position(latLng)
+        options.title(name)
+        options.icon(BitmapDescriptorFactory.fromResource(R.drawable.helper_marker))
 
-        // TODO Launch helper search
-        // a: Retrieve nearby users from online database            => loading bar (up to 50%
-        // b: send notification to users                            => loading bar (50-100%)
-        // c: wait for users                                        => spinning logo
-        // d: show users who accepted the help request on the map
+        mapsFragment.addMarker(options)
+    }
 
-        // TODO Pop up suggesting to call emergencies, explaining help is not assured
+    /**
+     * @see InfoOnHelpTaskActivity.setupLocation
+     */
+    private fun setupLocation() {
+        val currentLocation = GeneralLocationManager.get().getCurrentLocation(this)
+        if (currentLocation != null) {
+            currentLat = currentLocation.latitude
+            currentLong = currentLocation.longitude
+        } else {
+            // In case the permission to access the location is missing
+            val intent = Intent(this, MainPageActivity::class.java)
+            startActivity(intent)
+        }
+    }
+
+
+
+    /**
+     *  Opens the phone call app with the emergency number from the country the
+     *  user is currently in dialed. Called either from the popup or the button.
+     */
+    fun emergencyCall(view: View) {
+        val emergencyNumber =
+            LocalEmergencyCaller.getLocalEmergencyNumber(
+                currentLong,
+                currentLat,
+                this
+            )
+
+        val dial = "tel:$emergencyNumber"
+        startActivity(Intent(Intent.ACTION_DIAL, Uri.parse(dial)))
+    }
+
+    /**
+     * Auxiliary function to redirect the buttons presses to a tutorial page
+     */
+    private fun goToActivity(activityName: Class<*>?) {
+        val intent = Intent(this, activityName)
+        startActivity(intent)
+    }
+
+    /** Called when the user taps the asthma attack button */
+    fun goToAsthmaActivity(view: View) {
+        goToActivity(AsthmaActivity::class.java)
+    }
+
+    /** Called when the user taps the defibrillator button */
+    fun goToAedActivity(view: View) {
+        goToActivity(AedActivity::class.java)
+    }
+
+    /** Called when the user taps the allergies button */
+    fun goToAllergyActivity(view: View) {
+        goToActivity(AllergyActivity::class.java)
+    }
+
+    /** Called when the user taps the heart attack button */
+    fun goToHeartAttackActivity(view: View) {
+        goToActivity(HeartAttackActivity::class.java)
+    }
 
+    /**
+     * Cancels the search on the Database and goes back to MainActivity
+     */
+    fun cancelHelpSearch(view: View){
+        // TODO the action on the DB is not yet defined
```

Comment:
I know we discussed this already, but if you can find a way to somehow not end with 100 MainActivities this would be great.